kdc/kx509.c

```diff
@@ -347,16 +300,17 @@ kdc_kx509_verify_service_principal(krb5_context context,
                KRB5_TGS_NAME) == 0) {
         const char *r = krb5_principal_get_comp_string(context, sprincipal, 1);
         if ((ret = is_local_realm(context, reqctx, r)))
-            kx509_log(context, reqctx, 2, "client used wrong krbtgt for kx509");
+            _kdc_audit_addreason((kdc_request_t)reqctx,
+                                 "Client used wrong krbtgt for kx509");
         goto out;
     }
 
     /* Must be hostbased kca_service name then */
     ret = gethostname(localhost, sizeof(localhost) - 1);
     if (ret != 0) {
         ret = errno;
-        krb5_set_error_message(context, ret,
-                               N_("Failed to get local hostname", ""));
+        _kdc_audit_addreason((kdc_request_t)reqctx,
+                             "Failed to get local hostname");
```

Comment:
This one is perhaps high-severity local error (level 0) rather than a routine logging element for a particular request... Do we have access to kdc_log() here?

Reply:
I do.